lib/ansible/utils/plugin_docs.py

```diff
@@ -230,13 +236,16 @@ def get_docstring(filename, fragment_loader, verbose=False, ignore_errors=False,
             add_collection_to_versions_and_dates(data['doc'], collection_name, is_module=is_module)
 
         # add fragments to documentation
-        add_fragments(data['doc'], filename, fragment_loader=fragment_loader, is_module=is_module)
+        add_fragments(data['doc'], filename, fragment_loader=fragment_loader, is_module=is_module, section='DOCUMENTATION')
 
     if data.get('returndocs', False):
         # add collection name to versions and dates
         if collection_name is not None:
             add_collection_to_versions_and_dates(data['returndocs'], collection_name, is_module=is_module, return_docs=True)
 
+        # add fragments to return
+        add_fragments(data['returndocs'], filename, fragment_loader=fragment_loader, is_module=is_module, section='RETURN')
```

Comment:
I'm a bit worried that this will lead to sudden rejection of modules/plugins which have a return value called `extends_documentation_fragment` (while that would be quite surprising, it's technically allowed). How about doing something like:
```suggestion
        if not isinstance(data['returndocs'].get('extends_documentation_fragment'), Mapping):
            add_fragments(data['returndocs'], filename, fragment_loader=fragment_loader, is_module=is_module, section='RETURN')
```
Maybe with some warning/deprecation if `extends_documentation_fragment` turns out to be a mapping?